arch/x86/kvm/pkvm/mmu.c

```diff
@@ -180,6 +180,18 @@ static int fix_hyp_mmu_page_refcnt(void)
 	return pkvm_pgtable_walk(&hyp_mmu, 0, size, &walker);
 }
 
+static void set_host_mem_pgstate(unsigned long phys, unsigned long size,
+				 enum pkvm_page_state pgstate)
+{
+	unsigned long end = PAGE_ALIGN(phys + size);
```

Comment:
Hmm, what happens if phys or size is not page-aligned?

Should we at least warn about that (rather than just silently align them)?

Reply:
> In what cases would we want that?

Hmm, I don't know in what cases as this is just an assumption.

> ...What I was actually thinking about (not just in relation to set_host_mem_pgstate()): what exactly happens if any of the memblock regions (i.e. any of the regions in pkvm_memory) are not page-aligned? What would be the consequences?

We just treat it as page-aligned to map these regions in hyp mmu and host mmu. If they are not page-aligned, then means we map more in mmu. This sounds ok as the mappings in mmu has to be page-size granularity anyway?